Add Greatest and Least post aggregations

erikdubbelboer opened the pull request:
Adds `doubleGreatest`, `doubleLeast`, `longGreatest` and `longLeast` post aggregations which take a list of fields/aggregations and return the biggest or smallest value.

I was thinking of calling them `doubleMax` and `doubleMin` but this might be confusing with the already existing aggregations by that name which work a bit differently.

That's why I chose the name `Greatest` and `Least` like the [MySQL](http://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_greatest) and [PostgreSQL](https://www.postgresql.org/docs/8.4/static/functions-conditional.html#AEN15322) functions.

jon-wei commented:
can you add docs? looks good otherwise

erikdubbelboer commented:
@jon-wei I rebased on master and added some documentation.

fjy commented on docs/content/querying/post-aggregations.md:
````diff
@@ -54,6 +54,21 @@ The constant post-aggregator always returns the specified value.
 { "type"  : "constant", "name"  : <output_name>, "value" : <numerical_value> }
 ```
 
+### Greatest / Least post-aggregators
````
Can we call these max/min to match the naming convention of aggregators?

erikdubbelboer replied:
Like I mentioned I choose the name `Greatest` and `Least` like the [MySQL](http://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_greatest) and [PostgreSQL](https://www.postgresql.org/docs/8.4/static/functions-conditional.html#AEN15322) functions. SQL also has MAX and GREATEST which are almost the same but different. This also opens up the option to introduce these greatest/least functions as normal aggregators in the future.

jon-wei replied:
let's add a note to the docs that brings attention to the difference between MAX and GREATEST

jon-wei commented on processing/src/test/java/io/druid/query/aggregation/post/DoubleLeastPostAggregatorTest.java:
```diff
@@ -0,0 +1,94 @@
+            )
+        );
+
+    greatestPostAggregator = new DoubleLeastPostAggregator("least", postAggregatorList);
```
should be "leastPostAggregator", in testComparator() as well

jon-wei commented on processing/src/test/java/io/druid/query/aggregation/post/LongLeastPostAggregatorTest.java:
```diff
@@ -0,0 +1,94 @@
+            )
+        );
+
+    greatestPostAggregator = new LongLeastPostAggregator("least", postAggregatorList);
```
should be "leastPostAggregator", in testComparator() as well

erikdubbelboer commented:
@jon-wei I fixed those names (no idea how that happened). I added some explanation about the difference between `doubleMax` and `doubleGreatest`. And I rebased on master.

erikdubbelboer commented:
Any update on this? Would be awesome to get this into 0.9.3.

fjy commented:
👍

jon-wei commented:
:+1:

jon-wei commented:
@erikdubbelboer thanks for the contribution!

erikdubbelboer commented:
@jon-wei thank you for merging this. Looks like 0.9.3 will be the first release we will be able to use without modifications :)
